### ISSUE TITLE
Want to know more about the details of the cross validation method during training process

### ISSUE THREAD

**Issue by kevinco27** [sharathadavanne/seld-net#3]
According to the statement of SectionIII-D1 in paper and the code that only take one split of a dataset (e.g. **ANSYN_ov2_split1**) as input in a training procedure, I assume that you trained models on three split of a dataset separately, and finally averaged three outcomes obtaining from each model to determine the performance of model structure on a dataset. 

If my understanding about the training process mentioned above is correct, I wonder that **why you choose to treat each split of a dataset as independent ?** As my knowledge of cross validation, there are different methods to do it. And k-fold method, which takes two splits as training data and the rest one as testing data, maybe is more suitable for the small datasets in this paper, which have 150mins audios in one split. 

So, could you explain more about the cross validation in training procedure ? Thanks in advance.

**Comment by sharathadavanne** [comment]
Hi @kevinco27, that's a great question. First of all, thanks for taking the time to go through the paper and the code in detail. 

_**TLDR;** the datasets in the current work follow the cross-validation (CV)  terminology exactly if you consider only the task of sound events classification. But for the SELD task, since we are jointly estimating the sound event, it's onset-offset time and spatial location, there is no straightforward way to perform the CV split. The proposed way is one of the approaches for this multi-task CV splits_

I agree that the CV in our paper is not completely true to the definition of it. So let me explain why we are doing it differently. The task we are trying to solve is to recognize both the sound event and the spatial location of it.  For sound events, it is straightforward to set up a CV split - you basically split the examples for a particular sound class into train:test split. For example, if you have 20 examples for the sound event 'car', you randomly choose 4 for test split and use the remaining 16 for train split (80:20 splits). But there is no such straightforward approach for localization data. For example, if you treat the localization as a classification task, where each '(azimuth angle, elevation angle)' pair will be one class. In our implementation, we additionally place the sound event at different distances from the microphone. So for each '(azi, ele)' class, we have multiple examples of sound event and the distance that are varying. So one way to split the data for CV of localization is to have a disjoint set of sound events and distance for the train:test split. 
And that is exactly what we are doing (Section III.A. paragraph 3 of our [previous paper](https://arxiv.org/pdf/1710.10059.pdf)). We first create the three CV train and test split of sound events.  Up to this point, we follow the real definition of CV (Although in an ideal CV case, if we are using 80:20 split, we should have had 5 CV to cover all the examples equally, instead of 3 CV as used in the paper). Now for each of the three CV set of sound events, we generate the respective train/test split for the DOA task. But since for the SELD task we are synthesizing recordings with multiple sound events in it, that include the temporal information (onset and offset of each sound event). Each recording of 30s long that is generated will have N number of sound events from different classes, each with different spatial locations.  This restricts us from using the recording from one CV set in aother CV set, since the sound events part of the recording may not be part of the same train/test split in another CV. Due to this constraint, our CV splits for SELD task have different recordings from each other, and hence in the implementation you have three different datasets for the three splits.  

By the way, if we had a sufficiently large isolated sound events dataset, we could have just used one train/test split. But since there are no such publicly available large isolated sound events dataset, we use the 11 class dataset with 20 examples each which amounts to only 220 examples. Splitting this further into train:test splits will reduce the numbers. Hence we have to use some kind of CV to estimate the generalizing power of the classifier.
